nova_vm/src/ecmascript/builtins/promise/data.rs

```diff
@@ -8,23 +8,36 @@ use crate::{ecmascript::{builtins::control_abstraction_objects::promise_objects:
 pub struct PromiseHeapData {
     pub(crate) object_index: Option<OrdinaryObject>,
     pub(crate) promise_state: PromiseState,
-    pub(crate) promise_fulfill_reactions: Option<PromiseReactions>,
-    pub(crate) promise_reject_reactions: Option<PromiseReactions>,
-    pub(crate) promise_is_handled: bool,
 }
 
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Clone)]
 pub(crate) enum PromiseState {
-    #[default]
-    Pending,
+    Pending {
+        fulfill_reactions: Option<PromiseReactions>,
+        reject_reactions: Option<PromiseReactions>,
+        /// True if the resolution state of this promise depends on another
```

Comment:
**thought:** I don't quite understand when this gets used. Is it something like `Promise(async res => { ... })` where the outer Promise itself does "resolve" but it resolves into a Promise that is then not yet resolved?

Reply:
Yes, that's right.